src/LegacyPaymentGateways/Adapters/LegacyPaymentGatewayAdapter.php

```diff
@@ -36,12 +39,17 @@ public function getLegacyFormFieldMarkup(
         array $args,
         PaymentGatewayInterface $registeredGateway
     ): string {
-        return $registeredGateway->getLegacyFormFieldMarkup($formId, $args);
+        if (method_exists($registeredGateway, 'getLegacyFormFieldMarkup')) {
+            return $registeredGateway->getLegacyFormFieldMarkup($formId, $args);
+        }
+        
+        return false;
```

Comment:
The function return type is `string`, but we're returning `false` here. That will cause a type error. We probably want to return an empty string? Or perhaps throw an exception?

Reply:
you can actually return false on a string type, but I think I can elevate this check up a level

Reply:
moved this up a level to prevent hooking into action a31ffb66

Reply:
That's only true if strict types aren't enabled, in which case it will coerce the value.